avocado/core/job.py

```diff
@@ -499,7 +504,7 @@ def _run(self):
         self._log_job_debug_info(mux)
         replay.record(self.args, self.logdir, mux, self.urls)
         replay_map = getattr(self.args, 'replay_map', None)
-        summary = self.test_runner.run_suite(test_suite, mux, self.timeout,
+        summary = self.test_runner.run_suite(self.test_suite, mux, self.timeout,
                                              replay_map,
                                              self.args.test_result_total)
```

Comment:
OK I agree this is a step into the right direction and the implementation looks sound. Anyway you're opening the Pandora box here so I created a card https://trello.com/c/XvQ1NBhG/676-cleanup-the-job-class-as-it-contains-dead-code-and-methods-with-unnecessary-arguments and I really hope we'll have time to work on it in the next sprint to remove all unnecessary code.